python/unitytrainers/policy.py

```diff
@@ -0,0 +1,117 @@
+import logging
+
+from unityagents import UnityException
+from unitytrainers.models import LearningModel
+
+logger = logging.getLogger("unityagents")
+
+
+class UnityPolicyException(UnityException):
+    """
+    Related to errors with the Trainer.
+    """
+    pass
+
+
+class Policy(object):
```

Comment:
I really like the Policy abstraction! 

This comment is a high-level one about PPOPolicy and BCPolicy: they both contain feed_dict manipulations that are: 1. somewhat hard to parse, particularly as they appear in sequence, 2. common across both classes, 3. Use string literals (e.g.  'prev_action') that should really be class constants.

I think it would be much easier to parse if we had a class that was responsible for generating a feed_dict object and that class would leverage a series of static helper methods that update a feed_dict, the common ones can live in Policy and the algorithm-specific ones can live in their respective classes.

Listing sample common ones I've noticed, but there could be more:

```
for i, _ in enumerate(brain_info.visual_observations):
            feed_dict[self.model.visual_in[i]] = brain_info.visual_observations[i]
```

```
if self.use_vec_obs:
            feed_dict[self.model.vector_in] = brain_info.vector_observations
```

```
if not self.use_continuous_act:
            feed_dict[self.model.action_masks] = brain_info.action_masks
```